FBRetainCycleDetector/FBRetainCycleUtils.m

```diff
@@ -19,9 +19,27 @@
 #import "FBObjectiveCObject.h"
 #import "FBObjectGraphConfiguration.h"
 
-FBObjectiveCGraphElement *FBWrapObjectGraphElementWithContext(id object,
+static BOOL _ShouldBreakGraphEdge(FBObjectGraphConfiguration *configuration,
+                                  FBObjectiveCGraphElement *fromObject,
+                                  NSString *byIvar,
+                                  Class toObjectOfClass) {
+  for (FBGraphEdgeFilterBlock filterBlock in configuration.filterBlocks) {
+    if (filterBlock(fromObject, byIvar, toObjectOfClass) == FBGraphEdgeInvalid) {
+      return YES;
+    }
+  }
+
+  return NO;
+}
+
+FBObjectiveCGraphElement *FBWrapObjectGraphElementWithContext(FBObjectiveCGraphElement *sourceElement,
+                                                              id object,
                                                               FBObjectGraphConfiguration *configuration,
                                                               NSArray<NSString *> *namePath) {
+  if (_ShouldBreakGraphEdge(configuration, sourceElement, [namePath objectAtIndex:0], object_getClass(object))) {
```

Comment:
`objectAtIndex:` will crash (raise an exception) if the index is out of bounds. `firstObject` will return `nil` instead of crashing, so might wanna switch. I haven't backtracked to see where `namePath` comes from, maybe it's guaranteed to always have non-zero count.